File: src/util/log.ts
const DEBUG = process.env.POSTHOOK_DEBUG === "1";

export function info(msg: string): void {
  process.stderr.write(`${msg}\n`);
}

export function debug(msg: string): void {
  if (DEBUG) process.stderr.write(`[posthook] ${msg}\n`);
}

export function warn(msg: string): void {
  process.stderr.write(`[posthook] warning: ${msg}\n`);
}

File: internal/logx/log.go
package logx

import (
	"fmt"
	"os"
)

func debugEnabled() bool { return os.Getenv("POSTHOOK_DEBUG") == "1" }

func Info(msg string)                    { fmt.Fprintln(os.Stderr, msg) }
func Infof(format string, args ...any)   { fmt.Fprintln(os.Stderr, fmt.Sprintf(format, args...)) }
func Warn(msg string)                    { fmt.Fprintln(os.Stderr, "[posthook] warning: "+msg) }
func Warnf(format string, args ...any)   { fmt.Fprintln(os.Stderr, "[posthook] warning: "+fmt.Sprintf(format, args...)) }

func Debug(msg string) {
	if debugEnabled() {
		fmt.Fprintln(os.Stderr, "[posthook] "+msg)
	}
}

func Debugf(format string, args ...any) {
	if debugEnabled() {
		fmt.Fprintln(os.Stderr, "[posthook] "+fmt.Sprintf(format, args...))
	}
}
